### PR TITLE
Admin/fix basic issues

### PR DESCRIPTION
**Pull request by FrozenKiwi**
Admin app is slightly less broken.

dev & dev:live correctly connected.  Refactored mocking behaviour for webpack to common location (beside node_mocks, so they can keep in sync!)

### CODE REVIEW

File: apps/admin/src/containers/BrokerCAD/Clients/ClientSelect.tsx [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
 import { useFxRates } from "@thecointech/shared/containers/FxRate";
-import { UserState } from "./types";
-import { getAllUserData } from "./data";
+import { AllDataArray, getAllUserData } from "./data";
+import { AccountMap } from '@thecointech/shared/containers/AccountMap';
 
 export const ClientSelect = () => {
 
-  const [users, setUsers] = useState([] as UserState[]);
+  const [users, setUsers] = useState<AllDataArray>([]);
```
Comment (MarieTheCoin) [inline apps/admin/src/containers/BrokerCAD/Clients/ClientSelect.tsx:10]: AllDataArray is that the new any?

Comment (FrozenKiwi) [reply apps/admin/src/containers/BrokerCAD/Clients/ClientSelect.tsx:10]: hahaha!!! I could name that better, couldn't I?

File: apps/admin/webpack.renderer.config.js [hunk 2]
```diff
@@ -20,13 +20,18 @@ rules.push(
...
 );
 
-const mocked = require('./webpack.mocked.config');
+// In development mode, we mock most external libraries
+const mockOptions = process.env.NODE_ENV !== 'production'
+  ? require('../../__mocks__/mock_webpack')
+  : {};
+
 const baseOptions = {
   module: {
```
Comment (MarieTheCoin) [inline apps/admin/webpack.renderer.config.js:29]: What is that file for?

File: libs/firestore/src/types.ts [hunk 3]
```diff
@@ -14,7 +14,7 @@ export type FirestoreDataConverter<T> = firebase.firestore.FirestoreDataConverte
 export type DocumentData = firebase.firestore.DocumentData|firestore.DocumentData;
 export type SetOptions = firebase.firestore.SetOptions|firestore.SetOptions;
 
-// Typescript cannot workout both generics/overloads/union simultaneously
+// Typescript cannot work out both generics/overloads/union simultaneously
```
Comment (MarieTheCoin) [inline libs/firestore/src/types.ts:17]: The first version was also working :D

File: package.json [hunk 4]
```diff
@@ -50,6 +50,7 @@
     "storybook": "start-storybook -p 6006",
     "build-storybook": "build-storybook",
     "dedup": "yarn yarn-deduplicate -s fewer",
+    "dump-firestore": "cross-env CONFIG_NAME=devlive ts-node --project ./tsconfig.dev.json ./tools/dumpFirestore.ts",
```
Comment (MarieTheCoin) [inline package.json:53]: dump-firestore? What is this for?

**Review by MarieTheCoin — APPROVED** [review]

File: apps/admin/webpack.renderer.config.js (hunk 2 above)
Comment (FrozenKiwi) [reply apps/admin/webpack.renderer.config.js:30]: it's the single injection point.  If a webpack process merges that file, then all of the external services are automatically mocked

File: apps/admin/src/containers/BrokerCAD/Clients/ClientSelect.tsx (hunk 1 above)
Comment (FrozenKiwi) [reply apps/admin/src/containers/BrokerCAD/Clients/ClientSelect.tsx:10]: done in next PR

**Pull request merged**